src/mlpack/core/data/image_resize_crop.hpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file core/data/image_resize_crop.hpp
+ * @author Omar Shrit
+ *
+ * Image resize and crop functionalities.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+#define MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/stb/stb.hpp>
+
+namespace mlpack {
+namespace data {
+
+#ifndef MLPACK_DISABLE_STB
+
+/**
+ * Image resize/crop interfaces.
+ */
+
+/**
+ * Resize one single image matrix.
+ *
+ * @param image The input matrix that contains the image to be resized.
+ * @param info Contains relevant input images information.
+ * @param resizedImage The output matrix the contains the resized image.
+ * @param newWidth The new requested width for the resized image.
+ * @param newHeight The new requested height for the resized image.
+ */
+template<typename eT>
+inline void ResizeImage(arma::Mat<eT>& image, data::ImageInfo& info,
+    arma::Mat<eT> resizedImage, const size_t newWidth, const size_t newHeight)
```

Comment:
If we are modifying `info` in-place, then shouldn't be modify `image` in-place and not need the `resizedImage` parameter?